### PR TITLE
Translate to Japanese

### PR DESCRIPTION
**Pull request by panicdragon**
### Translate to Japanese

Nice to meet you, I am a front-end engineer panicdragon who works in Japan.

Your Questions are great.
I translated Japanese, so please merge it if you like.

### CODE REVIEW

**Comment by slumbers99** [comment]
@panicdragon ++

File: README-ja_JA.md [hunk 1]
```diff
@@ -0,0 +1,1335 @@
...
+<details><summary><b>答え</b></summary>
+<p>
+
+#### 答え: D
+
+関数内で、まず `var`キーワードを使って `name`変数を宣言します。これは、変数が定義されている行に実際に到達するまで、変数がデフォルト値の `undefined`で初期化される（作成時にメモリ空間が設定される）ことを意味します。 
+
+`name`変数をログ出力を実行している行では、まだ変数を定義していませんので、`undefined`の値を保持しています。
+
+`let`キーワード（または`const`）を持つ変数は持ち上げられますが、 `var`とは異なり、初期化されません。それらを宣言（初期化）する行の前にはアクセスできません。これは"temporal dead zone"と呼ばれます。
```
Comment (shingyoza) [inline README-ja_JA.md]: hoisting は「持ち上げ」ではなく「巻き上げ」を使ったほうがベターだと思われます

File: README-ja_JA.md [hunk 2]
```diff
@@ -0,0 +1,1335 @@
...
+- B: `20` と `NaN`
+- C: `20` と `63`
+- D: `NaN` と `63`
+
+<details><summary><b>答え</b></summary>
+<p>
+
+#### 答え: B
+
+`diameter`の値は正則関数であり、`perimeter`の値はアロー関数です。
```
Comment (shingyoza) [inline README-ja_JA.md]: 下の行と合わせて「通常の関数」ですかね

**Comment by lydiahallie** [comment]
Thanks so much for your translation! Could you add a link in the original `README.md` to the Japanese version?

**Comment by panicdragon** [comment]
Thank you for the review.
A Japanese link has been added.

**Pull request merged**